validator-server/src/test/kotlin/science/numcompass/conformance/fhir/chain/TerminologyServiceKtTest.kt

```diff
@@ -0,0 +1,76 @@
+package science.numcompass.conformance.fhir.chain
+
+import assertk.assertThat
+import assertk.assertions.isFalse
+import assertk.assertions.isTrue
+import org.junit.jupiter.api.Test
+
+internal class TerminologyServiceKtTest {
+
+    private val snomedSystemUrl = "http://snomed.info/sct"
+
+    @Test
+    fun `An arbitrary code not declared as SNOMED is not skipped`() {
+        val testString = "12344:4356=4545"
+        val valResult = isPostCoordinateSnomed("http://loinc.org", testString)
+        assertThat(valResult).isFalse()
```

Comment:
For me the ".. is not skipped" is a bit confusing.

After thinking about it, I understand that: `valResult` is `false`, so this means it either has been skipped or the regex matches.
Since the regex is not matching, it must have been skipped.

But actually looking at the code, I would say, it has been skipped since ´theCodeSystem` does not start with "http://snomed.info/sct".

What do you think about using different variables for the code, e.g.
```
val invalidSnomedCode = "12344:4356=4545"
val simpleSnomedCode = "12344"
```

Reply:
I agree that the naming is currently not optimal, so I will rework that. What I am actually checking is whether a (system, code)-pair looks like it's giving a SNOMED expression. It can fails that check equally by (i) not declared as a SNOMED-anything (system not SNOMED), or (ii) being declared a SNOMED-something but not looking like a complex expression.